vizro-core/src/vizro/models/_navigation/nav_item.py

```diff
@@ -0,0 +1,97 @@
+from __future__ import annotations
+
+import itertools
+from typing import Optional
+
+import dash
+import dash_bootstrap_components as dbc
+from dash import html
+from pydantic import validator
+
+from vizro._constants import STATIC_URL_PREFIX
+from vizro.models import VizroBaseModel
+from vizro.models._models_utils import _log_call
+from vizro.models._navigation._navigation_utils import _validate_pages
+from vizro.models._navigation.accordion import Accordion
+from vizro.models.types import NavPagesType
+
+
+class NavItem(VizroBaseModel):
+    """Icon to be used in Navigation Panel of Dashboard.
+
+    Args:
+        tooltip (Optional[str]): Text to be displayed in the tooltip on icon hover.
+        icon (Optional[str]): URI (relative or absolute) of the embeddable content.
+        pages (NavPagesType): See [NavPagesType][vizro.models.types.NavPagesType].
+    """
+
+    tooltip: Optional[str]
+    icon: Optional[str]
+    pages: NavPagesType
+    selector: Optional[Accordion] = None
+    text: Optional[str] = ""
+    max_text_length: int = 9
```

Comment:
I would change the order a bit to have obligatory fields first - this is relevant for the validation for example
Could you also add a field description to each where relevant?

```suggestion
    pages: NavPagesType
    selector: Optional[Accordion] = None
    icon: Optional[str]
    text: str = ""
    tooltip: Optional[str]
    max_text_length: int = 9
```

Reply:
Implemented here: https://github.com/mckinsey/vizro/pull/70/commits/222b583617a448a7162ac4339e3bae5f89a6e89d